### PR TITLE
Feature/10307 risk calculation http client/10310 request fetch hours

### PR DESCRIPTION
**Pull request by KaiTeuber**
## Description
request fetch hour migrated to new client
added a service helper to model fetchHours methodd
wifi rest service needed a special option to disable requests in developer menu

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-10310

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/HTTPClientCore/Service/Services/WifiOnlyRestService.swift [hunk 1]
```diff
@@ -15,24 +15,58 @@ class WifiOnlyRestService: Service {
...
+		let configuration: URLSessionConfiguration = wifiOnly ?
+			.coronaWarnSessionConfigurationWifiOnly() :
+			.coronaWarnSessionConfiguration()
+
+		return URLSession(configuration: configuration)
+	}
+
+	private var disabled = Set<String>()
+
+	func isDisabled(_ identifier: String) -> Bool {
```
Comment (ArturFriesen) [inline src/xcode/ENA/ENA/Source/HTTPClientCore/Service/Services/WifiOnlyRestService.swift:41]: Should this func also be inside of '#if !RELEASE' ?

File: src/xcode/ENA/ENA/Source/HTTPClientCore/Service/Services/_Service+Default.swift [hunk 2]
```diff
@@ -47,6 +47,12 @@ extension Service {
 		_ completion: @escaping (Result<R.Receive.ReceiveModel, ServiceError<R.CustomError>>) -> Void
 	) where R: Resource {
 
+		// check if resource loading might be disabled
+		if isDisabled(R.identifier) {
```
Comment (ArturFriesen) [inline src/xcode/ENA/ENA/Source/HTTPClientCore/Service/Services/_Service+Default.swift:51]: Should this call be inside of '#if !RELEASE' ?

File: src/xcode/ENA/ENA/Source/HTTPClientCore/Service/Services/_Service+Default.swift [hunk 3]
```diff
@@ -225,7 +238,7 @@ extension Service {
 		switch urlRequest(resource.locator, resource.sendResource, resource.receiveResource) {
 		case let .failure(resourceError):
 			Log.error("Creating url request failed.", log: .client)
-			retryOrDefaultValueOrFailureHandling(resource, .invalidRequestError(resourceError), nil, completion)
+			retryOrDefaultValueOrFailureHandling(resource, 0, .invalidRequestError(resourceError), nil, completion)
```
Comment (ArturFriesen) [inline src/xcode/ENA/ENA/Source/HTTPClientCore/Service/Services/_Service+Default.swift:241]: 0 is already default, not need to pass it here. (also for all other places)

**Review by ArturFriesen — CHANGES_REQUESTED** [review]

**Review by pascal-brause — COMMENTED** [review]
Is there the possibility to rename `PackageDownloadResponse` to something according to fetchHoursReceiveModel or similar?
And the `PackageDownloadResponse` to fetchHoursReceiveResource or similar?

File: src/xcode/ENA/ENA/Source/HTTPClientCore/Service/Services/_Service+Default.swift (hunk 3 above)
Comment (KaiTeuber) [reply src/xcode/ENA/ENA/Source/HTTPClientCore/Service/Services/_Service+Default.swift:241]: this only works with properties without the _ - I changed it

File: src/xcode/ENA/ENA/Source/Client/HTTP Client/__tests__/HTTPClient+DaysAndHoursTests.swift [hunk 4]
```diff
@@ -84,41 +83,47 @@ final class HTTPClientDaysAndHoursTests: CWATestCase {
 			responseData: Data("hello world".utf8)
```
[the remark this answers is not in the document]
Comment (KaiTeuber) [reply src/xcode/ENA/ENA/Source/Client/HTTP Client/__tests__/HTTPClient+DaysAndHoursTests.swift:83]: it gets move in the next PR. I kept it here until all old stuff is removed

File: src/xcode/ENA/ENA/Source/HTTPClient/Resources/FetchHour/FetchHoursServiceHelper.swift [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+	// MARK: - Internal
+
+	func fetchHours(
+		_ hours: [Int],
+		day: String,
+		country: String,
+		completion completeWith: @escaping (HoursResult) -> Void
+	) {
+		var errors = [Client.Failure]()
```
[the remark this answers is not in the document]
Comment (KaiTeuber) [reply src/xcode/ENA/ENA/Source/HTTPClient/Resources/FetchHour/FetchHoursServiceHelper.swift:25]: I removed an old type alias and more the rest over to the core

**Review by ArturFriesen — APPROVED** [review]

**Review by pascal-brause — APPROVED** [review]

**Pull request merged**